### PR TITLE
Open-source package licences page

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
Added an open-source package licences page which loads the data for each release from a static JSON file hosted on assets.

## QA
- VIsit https://ubuntu-com-11604.demos.haus/legal/open-source-licences
- Check that the first navigation item is selected and the table is populated
- Select other releases and check the navigation works and the table works when the json is correct.
- Impish and Focal desktop have broken source data at the moment

### CODE REVIEW

**Comment by anthonydillon** [comment]
### UX review
**Desktop**

- Releases are by development codename only, should we include the full name (Ubuntu 22.04 LTS Jammy Jellyfish)?
- Very long list - could we use pagination?
- Some basic search or filter would be good - I can take a look if you think it's worth it?
- If we don't implement filters, we should avoid pagination or make sure to include a “show all” in the Show “x” items drop down. If everything is on the screen the user can Ctrl+F. It’s a technical audience so they’ll likely do that. That could be good enough for MVP.

**Mobile**

- The “Toggle side navigation” puts an overlay over the screen, but doesn't offer any options, and I can't get out of it (I'm guessing that's a bug)
- 
- The cells are cut off - might need to use the .p-table--mobile-card class to make it work? I don't know how many people would look at this on a mobile screen, I'm guessing low, so might be OK for MVP? We should do it if it's not difficult.

**Comment by anthonydillon** [comment]
@ThinGuy have you had a chance to work on the JSON files? This branch is still waiting for them.

**Comment by anthonydillon** [comment]
@ThinGuy when the new JSON files are ready can you comment here and ping @sowasred2012 so someone on the team and complete and release this project.

Known remaining tasks for the developer that picks this up:
1. Upload the new JSON files to the asset server
2. Update the `data-file` attribute on the navigation items (https://github.com/canonical/ubuntu.com/pull/11604/files#diff-c075bcff4a522ca01a2967e259ef5e8dfa38566aae0f9bb69a87d8db618d75e2R33)
3. Check the navigation items now that all work and the links to the licences work as expected

**Comment by carkod** [comment]
@ThinGuy are the JSON files ready?

**Comment by anthonydillon** [comment]
Meet Craig and he said he would have the updated files over to me by tomorrow.

**Comment by anthonydillon** [comment]
@sowasred2012 nope, chasing again.

**Comment by anthonydillon** [comment]
Katherine +1'ed via email

**Review by steverydz — APPROVED** [review]
LGTM 👍

File: templates/legal/open-source-licences.html [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+  }
+
+  function renderTableRows(packages) {
+    let rows = "";
+    packages.forEach(function(package) {
+      rows += `<tr>
+        <td data-heading="Name">${package.name}</td>
+        <td data-heading="Version">${package.version}</td>
+        <td data-heading="Architecture">${package.component}</td>
+        <td data-heading="Licenses"><a href="${package.copyright_url}">${package.licenses}</a></td>
```
Comment (steverydz) [inline templates/legal/open-source-licences.html:187]: This outputs an empty link which was flagged by the WAVE Chrome extension

**Review by steverydz — CHANGES_REQUESTED** [review]
One comment in the code

**Comment by anthonydillon** [comment]
Good catch @steverydz. Fixed.

**Pull request merged**